pkg/kubelet/cm/topologymanager/policy_restricted.go

```diff
@@ -51,9 +51,26 @@ func (p *restrictedPolicy) canAdmitPodResult(hint *TopologyHint) lifecycle.PodAd
 	}
 }
 
+// Return hints that have Preferred equels true.
+func filterRestrictedHints(allResourcesHints [][]TopologyHint) [][]TopologyHint {
+	var filteredResourcesHints [][]TopologyHint
+	for _, oneResourceHints := range allResourcesHints {
+		var filtered []TopologyHint
+		for _, hint := range oneResourceHints {
+			if hint.Preferred == true {
+				filtered = append(filtered, hint)
+			}
+		}
+		filteredResourcesHints = append(filteredResourcesHints, filtered)
+	}
+	return filteredResourcesHints
+}
+
 func (p *restrictedPolicy) Merge(providersHints []map[string][]TopologyHint) (TopologyHint, lifecycle.PodAdmitResult) {
 	filteredHints := filterProvidersHints(providersHints)
-	hint := mergeFilteredHints(p.numaNodes, filteredHints)
```

Comment:
There's a lot of performance improvement potential in the topology manager algorithm for finding the NUMA nodes. It would be easier to review performance-related code if there was even a manual performance test for the TM algorithm. For example, have a fake machine with a lot of devices and NUMA nodes, and see how long it takes to find the best placement for workloads asking for various sets of devices. Do you think it would be a good idea to add such a check and then have some numbers how the filtering the preferred hints improves performance?

Reply:
We are currently developing a suite of tests for TM, aiming to be upstreamed for 1.18. I'm open to correction from @vpickard who is leading the efforts, but I don't think this type of scenario is currently being considered? 
Perhaps going forward and after we have the initial tests complete it would be useful to include something like this if possible.

Reply:
Ok, it's good to hear there is a testing effort underway.

Reply:
@ipuustin @nolancon The current focus is E2E testing to verify NUMA alignment on multi-NUMA systems, so this type of performance test isn't being addressed.

I like the idea, I don't think we will be able to address this in the 1.18 cycle, maybe afterwards.